lib/tbot/tbot_test.go

```diff
@@ -217,3 +351,46 @@ func tlsIdentFromDest(t *testing.T, dest bot.Destination) *tlsca.Identity {
 	require.NoError(t, err)
 	return tlsIdent
 }
+
+// TestBot_ResumeFromStorage ensures that after the bot stops, another instance
+// of the bot can be started from the state persisted to the storage
+// destination. This ensures that the renewable token join method will function
+// correctly.
+func TestBot_ResumeFromStorage(t *testing.T) {
```

Comment:
This test is intentionally pretty "loose", and just checking for no error. This means this test is more likely to outlive various refactors.